### PR TITLE
Feat/civ 6099 evidence upload main

### PR DESCRIPTION
**Pull request by filippo-possenti-version1**
### JIRA link (if applicable) ###
CIV-6099

### Change description ###
Evidence Upload functionality

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by drummondjm — APPROVED** [review]

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
   </module>
   <module name="SuppressWarningsFilter" />
   <module name="LineLength">
-    <property name="max" value="120"/>
+    <property name="max" value="180"/>
```
Comment (filippo-possenti-version1) [inline config/checkstyle/checkstyle.xml:37]: I'll make a note on this for other reviewers' benefit.

This is a change I made as TL. It's low-priority, doesn't affect the application logic and is within the domain of TLs, not devs.

Standard practice would be to have a separate ticket for this, so please keep in mind this is a bad practice I am adopting here and DON'T do it yourselves. I packed it in this ticket to see how reception of this change would have been.

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -212,7 +212,7 @@
       <property name="option" value="NL"/>
       <property name="tokens"
                 value="BAND, BOR, BSR, BXOR, DIV, EQUAL, GE, GT, LAND, LE, LITERAL_INSTANCEOF, LOR,
-                    LT, MINUS, MOD, NOT_EQUAL, PLUS, QUESTION, SL, SR, STAR, METHOD_REF "/>
+                    LT, MINUS, MOD, NOT_EQUAL, QUESTION, SL, SR, STAR, METHOD_REF "/>
```
Comment (filippo-possenti-version1) [inline config/checkstyle/checkstyle.xml:215]: I'll make a note on this for other reviewers' benefit.

This is a change I made as TL. It's low-priority, doesn't affect the application logic and is within the domain of TLs, not devs.

Standard practice would be to have a separate ticket for this, so please keep in mind this is a bad practice I am adopting here and DON'T do it yourselves. I packed it in this ticket to see how reception of this change would have been.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/EvidenceUploadRespondentNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyRespondentEvidenceUpload(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        String email = null;
+        if (isForRespondentSolicitor1(callbackParams)) {
+            email = caseData.getRespondentSolicitor1EmailAddress();
+        } else if (caseData.getAddRespondent2() != null
+                && caseData.getAddRespondent2().equals(YesOrNo.YES)
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/EvidenceUploadRespondentNotificationHandler.java:60]: Minor. Don't bother but keep it in mind.
you can probably swap the two sides of the second condition, which would result in handling both situations with just one condition:

`} else if (YesOrNo.YES.equals(caseData.getAddRespondent2() [...]`

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 4]
```diff
@@ -577,6 +581,54 @@ public boolean hasNoOngoingBusinessProcess() {
...
+    private final List<Element<UploadEvidenceWitness>> documentUploadWitness4Other;
+    private final List<Element<UploadEvidenceExpert>> documentUploadExpert1Other;
+    private final List<Element<UploadEvidenceExpert>> documentUploadExpert2Other;
+    private final List<Element<UploadEvidenceExpert>> documentUploadExpert3Other;
+    private final List<Element<UploadEvidenceExpert>> documentUploadExpert4Other;
+    private final List<Element<DocumentUploadTrial>> documentUploadTrial1Other;
+    private final List<Element<DocumentUploadTrial>> documentUploadTrial2Other;
+    private final List<Element<DocumentUploadTrial>> documentUploadTrial3Other;
+    private final List<Element<DocumentUploadTrial>> documentUploadTrial4Other;
+    private final LocalDateTime caseDocumentUploadDateOther;
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:631]: For other reviewers:
we are aware of the oddity around this whole structure. We'll be fixing this separately.

**Review by filippo-possenti-version1 — COMMENTED** [review]
Being the PR author I'm blocked from requiring changes. However, I'm not the author of the commits so I actually would like some changes in the code here.

Let's have a talk and let's see what we can improve before merging this code.

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by bhagyashreesharma90 — DISMISSED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by drummondjm — APPROVED** [review]

**Pull request merged**